includes/api/class-wc-admin-rest-products-controller.php

```diff
@@ -89,50 +98,67 @@ protected function prepare_objects_query( $request ) {
 	 * @return WP_Error|WP_REST_Response
 	 */
 	public function get_items( $request ) {
-		add_filter( 'posts_where', array( __CLASS__, 'add_wp_query_product_search_filter' ), 10, 2 );
-		add_filter( 'posts_join', array( __CLASS__, 'add_wp_query_product_search_join' ), 10, 2 );
-		add_filter( 'posts_groupby', array( __CLASS__, 'add_wp_query_product_search_group_by' ), 10, 2 );
+		add_filter( 'posts_where', array( __CLASS__, 'add_wp_query_filter' ), 10, 2 );
+		add_filter( 'posts_join', array( __CLASS__, 'add_wp_query_join' ), 10, 2 );
+		add_filter( 'posts_groupby', array( __CLASS__, 'add_wp_query_group_by' ), 10, 2 );
 		$response = parent::get_items( $request );
-		remove_filter( 'posts_where', array( __CLASS__, 'add_wp_query_product_search_filter' ), 10 );
-		remove_filter( 'posts_join', array( __CLASS__, 'add_wp_query_product_search_join' ), 10 );
-		remove_filter( 'posts_groupby', array( __CLASS__, 'add_wp_query_product_search_group_by' ), 10 );
+		remove_filter( 'posts_where', array( __CLASS__, 'add_wp_query_filter' ), 10 );
+		remove_filter( 'posts_join', array( __CLASS__, 'add_wp_query_join' ), 10 );
+		remove_filter( 'posts_groupby', array( __CLASS__, 'add_wp_query_group_by' ), 10 );
 		return $response;
 	}
 
 	/**
-	 * Allow searching by product name or sku in WP Query
+	 * Add in conditional search filters for products.
 	 *
 	 * @param string $where Where clause used to search posts.
 	 * @param object $wp_query WP_Query object.
 	 * @return string
 	 */
-	public static function add_wp_query_product_search_filter( $where, $wp_query ) {
+	public static function add_wp_query_filter( $where, $wp_query ) {
 		global $wpdb;
 
 		$search = trim( $wp_query->get( 'search' ) );
 		if ( $search ) {
 			$search = $wpdb->esc_like( $search );
 			$search = "'%" . $search . "%'";
-			$where .= ' AND (' . $wpdb->posts . '.post_title LIKE ' . $search;
+			$where .= " AND ({$wpdb->posts}.post_title LIKE {$search}";
 			$where .= wc_product_sku_enabled() ? ' OR ps_post_meta.meta_key = "_sku" AND ps_post_meta.meta_value LIKE ' . $search . ')' : ')';
 		}
 
+		if ( $wp_query->get( 'low_in_stock' ) ) {
+			$where .= " AND lis_postmeta2.meta_key = '_manage_stock' AND lis_postmeta2.meta_value = 'yes'
+			AND lis_postmeta.meta_key = '_stock' AND lis_postmeta.meta_value IS NOT NULL
+			AND (
+				lis_postmeta3.meta_key = '_low_stock_amount' AND lis_postmeta.meta_value IS NOT NULL
+				AND lis_postmeta3.meta_key = '_low_stock_amount' AND CAST(lis_postmeta.meta_value AS SIGNED) <= CAST(lis_postmeta3.meta_value AS SIGNED)
+				OR lis_postmeta3.meta_key = '_low_stock_amount' AND lis_postmeta.meta_value IS NULL
+				AND lis_postmeta.meta_key = '_stock' AND CAST(lis_postmeta.meta_value AS SIGNED) <= '{$low_stock_amount}'
```

Comment:
Looks like `$low_stock_amount` didn't make the jump from https://github.com/woocommerce/woocommerce-admin/pull/1866/commits/19c8228b4e87eceb905abb1b5429b2cd71dd2f11#diff-04ebb0cb4b33c92c166ab779a502d791R530.

Also, could we delete the first two commits?

Reply:
Whoops- thanks!

>Also, could we delete the first two commits?

Yep.  Wanted to make sure this was the right direction before dropping those.